test/it/helpers.js

```diff
@@ -2,31 +2,13 @@ var common = require('../common');
 var uuid = common.uuid;
 var stormpath = common.Stormpath;
 
-function loadApiKey(cb) {
-  var id = process.env.STORMPATH_CLIENT_APIKEY_ID;
-  var secret = process.env.STORMPATH_CLIENT_APIKEY_SECRET;
-  var homeDir = process.env[(process.platform === 'win32' ? 'USERPROFILE' : 'HOME')] || '';
-  var apiKeyFilePath = homeDir + '/.stormpath/apiKey.properties';
-
-  if (id && secret) {
-    return process.nextTick(function() {
-      cb(new stormpath.ApiKey(id, secret));
-    });
-  }
-
-  stormpath.loadApiKey(apiKeyFilePath, function(err, apiKey) {
-    if (err) {
-      throw err;
-    }
-
-    cb(apiKey);
-  });
-}
+var loadedClient = null;
 
 function getClient(cb) {
-  loadApiKey(function(apiKey) {
+  if (loadedClient) {
+    cb(loadedClient);
```

Comment:
I think it's bad for readability to encapsulate almost an entire function inside an `else` statement:

```javascript
function getClient(cb) {
  if (loadedClient) {
    cb(loadedClient);
  } else {
    var client = new stormpath.Client({
      skipRemoteConfig: true
    });

    client.on('error', function (err) {
      throw err;
    });

    client.on('ready', function () {
      loadedClient = client;
      cb(client);
    });
  }
}
```

**vs.**

```javascript
function getClient(cb) {
  if (loadedClient) {
    return cb(loadedClient);
  }

  var client = new stormpath.Client({
    skipRemoteConfig: true
  });

  client.on('error', function (err) {
    throw err;
  });

  client.on('ready', function () {
    loadedClient = client;
    cb(client);
  });
}
```